test/e2e/performanceprofile/functests/utils/cgroup/v1/v1.go

```diff
@@ -0,0 +1,101 @@
+package v1
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/cgroup/controller"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/cgroup/runtime"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/pods"
+)
+
+type ControllersManager struct {
+	client    client.Client
+	k8sClient *kubernetes.Clientset
+}
+
+func NewManager(c client.Client, k8sClient *kubernetes.Clientset) *ControllersManager {
+	return &ControllersManager{client: c, k8sClient: k8sClient}
+}
+
+func (cm *ControllersManager) CpuSet(ctx context.Context, pod *corev1.Pod, containerName, childName, runtimeType string) (*controller.CpuSet, error) {
+	cfg := &controller.CpuSet{}
+	dirPath := path.Join(controller.CgroupMountPoint, childName)
+	cmd := []string{
+		"/bin/cat",
+		dirPath + "/cpuset/cpuset.cpus",
+		dirPath + "/cpuset/cpuset.cpu_exclusive",
+		dirPath + "/cpuset/cpuset.effective_cpus",
+		dirPath + "/cpuset/cpuset.mems",
+		dirPath + "/cpuset/cpuset.sched_load_balance",
+	}
+	b, err := pods.ExecCommandOnPod(cm.k8sClient, pod, containerName, cmd)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve cgroup config for pod. pod=%q, container=%q; %w", client.ObjectKeyFromObject(pod).String(), containerName, err)
+	}
```

Comment:
This method of passing multiple interface files at once and reading them is problematic because there is no guarantee that all the files will be read, there have been cases where only the first file is read and others are skipped and no error is returned.  Or some times one of the files are read. 

because "cat"  when run on normal system and when run inside a container , the behavior changes and it can skip files. We had multiple instances in past where we have faced these issues. 

Also we are just using split to differentiate the outputs between different interface files , though the files are read in the same order as given, if for some reason one of files are not read or the values are empty due to some errors, the values can get mixed up causing the interface values in cfg struct  fields to be jumbled.

Reply:
+1. perhaps is better to divide this into submethods to fetch each value and reporting errors/warnings accordingly.

Reply:
I just did please review the last commit

Reply:
the latest changes look good . We can improve iteratively.

Reply:
LGTM from my side